### PR TITLE
BEAM-2546 Beam IO for InfluxDB

### PR DESCRIPTION
**Pull request by bipinupd**
Beam IO for InfluxDB. I am new to BEAM community and will appreciate the suggestion and feedback.
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @ jbonofre`). 
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by bipinupd** [comment]
`R: @ jbonofre`

**Comment by bipinupd** [comment]
R: @ kenn

**Comment by aaltay** [comment]
@jbonofre - would you be able to review this or suggest an alternative reviewer?

**Comment by aaltay** [comment]
cc: @kennknowles @chamikaramj -- for reviewer suggestions.

**Comment by kennknowles** [comment]
I think @rezarokni might be a great choice for reviewing IO for a time series DB.

**Comment by aaltay** [comment]
@rezarokni - would you be interested in reviewing this?

@bipinupd -- in the meantime if you have time, could you look at the failing tests?

**Comment by aaltay** [comment]
@iemejia -- maybe you can suggest a good reviewer here?

**Comment by bipinupd** [comment]
> @rezarokni - would you be interested in reviewing this?
> 
> @bipinupd -- in the meantime if you have time, could you look at the failing tests?

Thanks @aaltay. I think I was missing `infra` related to part for my integration test. I have added them.

**Comment by iemejia** [comment]
Oups forgot this one, I will take a look.

**Comment by bipinupd** [comment]
> Oups forgot this one, I will take a look.

Thanks @iemejia

File: sdks/java/io/influxdb/src/main/java/org/apache/beam/sdk/io/influxdb/InfluxDBIO.java [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+                connection.setDatabase(spec.database());
+            }
+
+            @StartBundle
+            public void startBundle(StartBundleContext c) {
+            }
+
+            @ProcessElement
+            public void processElement(ProcessContext c) {
+//                    Point point = Point.measurementByPOJO(c.element().getClass()).addFieldsFromPOJO(c.element()).build();
```
Comment (rezarokni) [inline sdks/java/io/influxdb/src/main/java/org/apache/beam/sdk/io/influxdb/InfluxDBIO.java]: Needed?

File: sdks/java/io/influxdb/src/main/java/org/apache/beam/sdk/io/influxdb/InfluxDBIO.java [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+                builder.delegate(Write.this);
+            }
+
+            private final Integer DEFAULT_NUMBER_OF_DURATION = 1000;
+            private final Integer DEFAULT_FLUSH_DURATION = 100;
+
+        }
+    }
+
+    public static OkHttpClient.Builder getUnsafeOkHttpClient() {
```
Comment (rezarokni) [inline sdks/java/io/influxdb/src/main/java/org/apache/beam/sdk/io/influxdb/InfluxDBIO.java]: @kennknowles Sorry, do you know what normal Beam policy is around dealing with certs is?

File: settings.gradle [hunk 3]
```diff
@@ -145,8 +149,8 @@ include ":sdks:python:test-suites:tox:py2"
 include ":sdks:python:test-suites:tox:py35"
 include ":sdks:python:test-suites:tox:py36"
 include ":sdks:python:test-suites:tox:py37"
-include ":vendor:grpc-1_21_0"
-include ":vendor:bytebuddy-1_9_3"
+include ":vendor:grpc-1_26_0"
+include ":vendor:bytebuddy-1_10_8"
```
Comment (alexvanboxel) [inline settings.gradle]: The vendored versions should not be changed in this PR. Probably better rebase this PR. Focus in the settings on changes for the PR

**Comment by bipinupd** [comment]
@iemejia I think I did something wrong while rebasing. It is showing way too much file changed, however the branch looks fine [only files related to my project are showing] (https://github.com/apache/beam/compare/master...bipinupd:BEAM-2546?expand=1) . 

Do you want me to open a new PR. Also, I did not find embedded/mock server or InfluxDB.

**Comment by iemejia** [comment]
Oups yes it looks pretty screwed up. You can fix those errors doing `git pull origin master --rebase` but you would have to deal with the merge conflicts, you can also open a new PR if you prefer just mention me there if you take that approach to continue the review there..

**Comment by bipinupd** [comment]
Closing this pull request. I did something wrong while rebasing. It is showing way too much file changed.

**Pull request closed without merging**